### PR TITLE
fix(clerk-js): Revalidate Organization Memberlist when joining an Organization

### PR DESCRIPTION
**Pull request by octoper**
## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [X] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [X] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Comment by octoper** [comment]
!preview

**Comment by clerk-cookie** [comment]
Hey @octoper, your preview is currently deploying and should be available shortly.

| Status | Preview | Updated (UTC) |
| :-- | :-- | :-- |
| :cook: _Deploying..._ | [Inspect](https://javascript-preview-qpp0o3dcw.clerkstage.dev) | Dec 21, 2023 06:07 PM |

**Review by anagstef — APPROVED** [review]

**Comment by panteliselef** [comment]
!preview

**Comment by panteliselef** [comment]
@anagstef could u review again, as i've built on top of the old solution ? It was a bit tricky to get it working without the layout shift when accepting an invitation, but it should be ok now

**Comment by octoper** [comment]
!preview

File: packages/clerk-js/src/ui/components/OrganizationList/OrganizationListPage.tsx [hunk 1]
```diff
@@ -121,6 +121,11 @@ const OrganizationListPageList = (props: { onCreateOrganizationClick: () => void
   const handleCreateOrganizationClicked = () => {
     props.onCreateOrganizationClick();
   };
+
+  // Solve weird bug with swr while running unit tests
+  const userInvitationsData = userInvitations.data?.filter(a => !!a);
+  const userSuggestionsData = userSuggestions.data?.filter(a => !!a);
```
Comment (anagstef) [inline packages/clerk-js/src/ui/components/OrganizationList/OrganizationListPage.tsx:127]: Generally, it's a bad practice to alter the way your code works in favour of the tests. 😿

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/InPlaceAcceptedInvitations.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import type { OrganizationResource, UserOrganizationInvitationResource } from '@clerk/types';
```
Comment (anagstef) [inline packages/clerk-js/src/ui/components/OrganizationSwitcher/InPlaceAcceptedInvitations.tsx:1]: This should be renamed and moved into the `ui/contexts` directory.

**Review by anagstef — APPROVED** [review]
💯  Just 2 small comments

File: packages/clerk-js/src/ui/components/OrganizationList/OrganizationListPage.tsx (hunk 1 above)
Comment (panteliselef) [reply packages/clerk-js/src/ui/components/OrganizationList/OrganizationListPage.tsx:127]: Totally agree, for some reason only in test and not runtime in a browser `swr` will populate data with `[undefined]` and `total_count` would still be `0`. I have not figured out why this is happening, I have added a note in order to come back later and investigate deeper.

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/InPlaceAcceptedInvitations.tsx (hunk 2 above)
Comment (panteliselef) [reply packages/clerk-js/src/ui/components/OrganizationSwitcher/InPlaceAcceptedInvitations.tsx:1]: Should we ? This is something that is needed only for OrganizationSwitcher

**Pull request merged**